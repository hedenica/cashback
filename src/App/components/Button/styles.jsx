import styled from 'styled-components';
import { shade } from 'polished';

export const Container = styled.button`
  background: #fbab8c;
  height: 56px;
  border-radius: 10px;
  border: 0;
  padding: 0 16px;
  color: #666360;
  width: 100%;
  font-weight: 500;
  margin-top: 16px;
  transition: background-color 0.5s;

  &:hover {
    background: ${shade(0.2, '#fbab8c')};
  }
`;
