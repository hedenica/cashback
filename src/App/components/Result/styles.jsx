import styled from 'styled-components';

export const Container = styled.div`
  text-align: center;
  padding: 50px;
`;

export const Image = styled.div`
  margin-bottom: 48px;
`;

export const Content = styled.div`
  h4 {
    text-align: center;
    margin-bottom: 32px;
  }
  p {
    width: 305px;
    font-size: 16px;
    text-align: center;
    margin: 10px auto;
  }
`;
