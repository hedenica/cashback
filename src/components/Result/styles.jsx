import styled from 'styled-components';

export const Container = styled.div`
  text-align: center;
  padding: 50px;
`;

export const Content = styled.div`
  h4 {
    text-align: center;
    max-width: 350px;
    margin: 10px auto;
  }

  p {
    width: 305px;
    font-size: 16px;
    text-align: center;
    margin: 10px auto;
  }

  button {
    width: 250px;
  }
`;
