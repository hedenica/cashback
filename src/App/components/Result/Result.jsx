import React from 'react';
import PropTypes from 'prop-types';

import { Container, Image, Content } from './styles';

const Result = ({ image, title, description, callToAction }) => (
  <Container>
    <Image>{image}</Image>
    <Content>
      <h4>{title}</h4>
      <p>{description}</p>
      {callToAction && callToAction}
    </Content>
  </Container>
);

export default Result;

Result.propTypes = {
  callToAction: PropTypes.node,
  description: PropTypes.node,
  image: PropTypes.node.isRequired,
  title: PropTypes.string,
};

Result.defaultProps = {
  callToAction: null,
  description: '',
  title: '',
};
