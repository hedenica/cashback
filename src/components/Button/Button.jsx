import React from 'react';
import PropTypes from 'prop-types';

import { Container } from './styles';

const Button = ({ children, ...rest }) => (
  <Container type="button" {...rest}>
    {children}
  </Container>
);

export default Button;

Button.propTypes = {
  children: PropTypes.node.isRequired,
};
