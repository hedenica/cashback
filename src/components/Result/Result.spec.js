import React from 'react';
import { render } from '@testing-library/react'

import Result from './Result';
import Button from '../Button'
import Image from '../../assets/img/logo.png';


describe('<Result />', () => {
  it('should render the result component with image', () => {
    const { container } = render (
      <Result image={Image} callToAction={
        <Button>
          Clique aqui
        </Button>
      } />
    )
    expect(container).toMatchSnapshot();
  });
});
