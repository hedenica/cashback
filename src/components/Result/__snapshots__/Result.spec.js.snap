// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Result /> should render the result component with image 1`] = `
<div>
  <div
    class="sc-bdfBwQ hPHeAY"
  >
    
    <div
      class="sc-gsTCUz bEPrJK"
    >
      <h4 />
      <p />
    </div>
  </div>
</div>
`;
